### PR TITLE
Update and enable Search component

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Implements Search component powered by DocSearch and Algolia
- Migrates/updates `sanitizeHitTitle`—no longer needs checks for escaped character (`&apos;`, etc)
- Update `.env.example` using `NEXT_PUBLIC_` prefix; remove deprecated variables, add some comments (env vars updated in Netlify as well)

## Related Issue
https://www.notion.so/efdn/Implement-search-functionality-bb7988a4cc8c4de6b42c4797b02d2269?pvs=4

### CODE REVIEW

**Review by pettinarip — APPROVED** [review]
Awesome! great job @wackerow. Nice to see this working on the fork.

File: .env.example [hunk 1]
```diff
@@ -1,17 +1,20 @@
 # rename this file to .env and supply the values listed below
 # also make sure they are available to the build tool (e.g. Netlify)
-# warning: variables prefixed with GATSBY_ will be made available to client-side code
+# warning: variables prefixed with NEXT_PUBLIC_ will be made available to client-side code
 # be careful not to expose sensitive data (e.g. your Algolia admin key)
-# ALGOLIA_ADMIN_KEY=insertValue
+
+# Algolia environment (app ID, search key and base search index name required for search)
+# NEXT_PUBLIC_ALGOLIA_APP_ID=insertValue
```
Comment (nhsz) [inline .env.example:7]: do we need the `ALGOLIA` vars to be prefixed by `NEXT_PUBLIC`?

**Review by nhsz — COMMENTED** [review]
LGTM, nice! 🙌 Left a comment about Algolia vars being public

File: .env.example (hunk 1 above)
Comment (wackerow) [reply .env.example:7]: We do, yes... these are consumed by the component when accessing the index. None of these variables are sensitive (just read-access to the index), and is parity to the canonical site where we use the `GATSBY_` prefix.

**Review by nhsz — APPROVED** [review]

**Pull request merged**